code/obj/machinery/computer/security.dm

```diff
@@ -33,8 +43,18 @@
 		user.set_eye(C)
 		return 1
 
+	proc/move_viewport_to_camera(var/obj/machinery/camera/C, client/clint)
+		var/datum/viewport/vp = clint.getViewportsByType("cameras: Viewport")[1]
+		var/turf/T = get_turf(C)
+		var/turf/closestPos = null
+		for(var/i = 4, i >= 0 || !closestPos, i--)
+			closestPos = locate(T.x - i, T.y + i, T.z)
+			if(closestPos) break
+		vp.SetViewport(closestPos, 8, 8)
+
 	//moved out of global to only be used in sec computers
-	proc/move_security_camera(/*n,*/direct,var/mob/living/carbon/user)
+	proc/move_security_camera(direct, client/clint)
+		var/mob/user = clint.mob
```

Comment:
Yeah I was just commenting on how you don't necessarily know that a human is interacting with it, you only know a mob is